pkg/appmgmt/general/general_test.go

```diff
@@ -526,3 +527,269 @@ func TestGetOwnerReferences(t *testing.T) {
 	assert.Equal(t, returnedOwnerRefs[0].Kind, "Pod", "Unexpected owner reference Kind")
 	assert.Equal(t, returnedOwnerRefs[0].APIVersion, v1.SchemeGroupVersion.String(), "Unexpected owner reference Kind")
 }
+
+// nolint: funlen
+func TestListApplication(t *testing.T) {
```

Comment:
This should be a table driven test.
We have 4 things that can change: Labels, Annotations, SchedulerName and NodeName. The outcome is known: app exists or not. Setting up a table with those 4 inputs and the expected output. Create a new pod lister or add a reset function on the mocked pod lister and run the checks in a loop.

Reply:
I agree with @wilfred-s. When possible we should use table driven tests. Is much easier to read, and it will be shorter as well, so we might not need to skip the lint funlen check, if we do, then maybe it worth to split the test instead of just ignoring it.